lib/GeometryUtilities.js

```diff
@@ -46,66 +43,71 @@ const getLongestEdgeIndex = bb => {
 // triangles in the geometry
 const getAverage = ( tris, avg, geo, getValFunc ) => {
 
-	avg.set( 0, 0, 0 );
+	let avgx = 0;
+	let avgy = 0;
+	let avgz = 0;
 
 	for ( let i = 0, l = tris.length; i < l; i ++ ) {
 
 		const tri = tris[ i ];
 
 		for ( let v = 0; v < 3; v ++ ) {
 
-			avg.x += getValFunc( geo, tri, v, 0 );
-			avg.y += getValFunc( geo, tri, v, 1 );
-			avg.z += getValFunc( geo, tri, v, 2 );
+			avgx += getValFunc( geo, tri, v, 0 );
+			avgy += getValFunc( geo, tri, v, 1 );
+			avgz += getValFunc( geo, tri, v, 2 );
 
 		}
 
 	}
 
-	avg.x /= tris.length * 3;
-	avg.y /= tris.length * 3;
-	avg.z /= tris.length * 3;
+	avg.x = avgx / ( tris.length * 3 );
+	avg.y = avgy / ( tris.length * 3 );
+	avg.z = avgz / ( tris.length * 3 );
 
 };
 
 // shrinks the provided bounds on any dimensions to fit
 // the provided triangles
 const shrinkBoundsTo = ( tris, bounds, geo, getValFunc ) => {
 
-	bndtemp.min.x = Infinity;
-	bndtemp.min.y = Infinity;
-	bndtemp.min.z = Infinity;
+	let minx = Infinity;
+	let miny = Infinity;
+	let minz = Infinity;
 
-	bndtemp.max.x = - Infinity;
-	bndtemp.max.y = - Infinity;
-	bndtemp.max.z = - Infinity;
+	let maxx = - Infinity;
+	let maxy = - Infinity;
+	let maxz = - Infinity;
+
+	let x, y, z;
 
 	for ( let i = 0, l = tris.length; i < l; i ++ ) {
 
 		const tri = tris[ i ];
 
 		for ( let v = 0; v < 3; v ++ ) {
 
-			const x = getValFunc( geo, tri, v, 0 );
-			const y = getValFunc( geo, tri, v, 1 );
```

Comment:
Is declaring new variables in a for loop body really a bottle neck? I didn't know that. Everything else looks good -- I'm just curious about this!

Reply:
I don't think this part mattered, I think the part that mattered was updating a local variable directly vs. updating properties on the local vector.